src/olympia/promoted/models.py

```diff
@@ -38,36 +38,30 @@ def all_for(self, addon):
     def approved_for(self, addon):
         if not addon.current_version:
             return self.none()
-        # For each PromotedGroup, the group should have an
-        # associated promoted_version version that is:
-        # 1. The addon's current version
-        # 2. The same promoted group
-        # 3. The group has approved_applications for the current version
-        # OR is not prereviewed for all apps
-        groups = (
-            self.all_for(addon=addon)
-            .filter(
-                Q(
-                    promoted_versions__version=addon.current_version,
-                    promotedaddonpromotion__promoted_group=models.F('id'),
-                )
-                | Q(
-                    promotedaddonpromotion__promoted_group__listed_pre_review=False,
-                    promotedaddonpromotion__promoted_group__unlisted_pre_review=False,
+
+        # An addon is approved for a promoted group if:
+        # 1. For each PromotedAddonVersion A, there exists a
+        #    PromotedAddonPromotion B (an approval) such that
+        #       i. Are the same group,
+        #       ii. Are the same application,
+        #       iii. A.version = B.addon.current_version
+        # OR, 2. is a promoted group that does not require prereview.
```

Comment:
remove the OR, the numbers read better when they are vertically aligned

Reply:
Also.. is that how that should be? So there will be cases where `approved_for` returns groups where there are no PromotedAddonPromotion instances? That feels wrong.

Reply:
See src/olympia/reviewers/tests/test_utils.py::TestReviewHelper::test_autoapprove_fails_for_promoted, specifically line #3213

Reply:
🤷 ok.